src/vs/platform/webview/common/webviewManagerService.ts

```diff
@@ -13,7 +13,7 @@ export const IWebviewManagerService = createDecorator<IWebviewManagerService>('w
 export interface IWebviewManagerService {
 	_serviceBrand: unknown;
 
-	registerWebview(id: string, webContentsId: number, metadata: RegisterWebviewMetadata): Promise<void>;
+	registerWebview(id: string, webContentsId: number | undefined, metadata: RegisterWebviewMetadata): Promise<void>;
```

Comment:
I need to figure out what the equivalent of `webContentsId` for iframe based webviews. We need this for intercepting the localhost request for remote

Reply:
@deepak1556 Here is the problem I am trying to solve:

1. For port forwarding in remote cases, we use a `onBeforeRequest` that looks at requests to `localhost` and potentially rewrites them

1. There is a single `onBeforeRequest` handler for all webviews

1. When using the `<webview>` tag, we were able to identify which webview was making the request to localhost using the web contents id. We would then use this id to lookup the port mapping information for that webview

1. However iframes do not have a unique web contents id (at least, not by default) I cannot find any way to map a request that comes in to `onBeforeRequest` back to the webview that triggered the request.

Here are the fields I see on the `details` object in `onBeforeRequest`

<img width="856" alt="Screen Shot 2020-06-24 at 5 01 02 PM" src="https://user-images.githubusercontent.com/12821956/85639372-657a5e00-b63d-11ea-9d98-a96a65394e5f.png">

Any thoughts on how we could handle this mapping?

Reply:
@mjbvz I can extend the webrequest api to provide some addtional event details that would include the frame origin and frame name, do you think its good enough to manage the port map ?

`vscode-webview://787e559f-185c-4d86-aebc-400191530e52/...`

`<iframe name="test" >`

then details will have

```
{
  isSubFrame: true, // if the request originated from a subframe
  origin: '787e559f-185c-4d86-aebc-400191530e52', // last committed origin of the frame
  name: 'test', // value of the name attribute for a subframe
  ....
}
```

Reply:
Yes that would work great for use. Should I open a feature request against electron?

Reply:
Yes please do, I will put up a PR targeting it.

Reply:
Great! Opened https://github.com/electron/electron/issues/24303 for this